packages/flutter/lib/src/rendering/object.dart

```diff
@@ -194,7 +196,7 @@ class PaintingContext extends ClipContext {
     assert(_canvas == null || _canvas!.getSaveCount() == 1);
 
     // Create a layer for our child, and paint the child into it.
-    if (child._needsPaint) {
+    if (child._needsPaint || child.layer == null) {
```

Comment:
In theory we might need this if you weren't a repaint boundary, but before markNeedsPaint you decide to become one